rtree/golden_test.go

```diff
@@ -139,7 +139,7 @@ func checksum(n *node) uint64 {
 	var entries []string
 	for i := 0; i < n.numEntries; i++ {
 		var entry string
-		if n.isLeaf {
+		if n.entries[i].child == nil {
```

Comment:
From what I noticed above, a leaf node is defined by `numEntries <= 4`. IIUC, The definition of leaf would then be different to `entry.child == nil`.

If that's the case, perhaps we can wrap this definition as a node method?

```go
func (n node) isLeaf() bool {
    // 4 or fewer entries can fit into a single node.
    return n.numEntries <= 4
}
```

But if that's not the case, perhaps still putting `func (e entry) isLeaf() bool { return e.child == nil }` might make it more readable, or would it be more confusing? 😅

Reply:
> From what I noticed above, a leaf node is defined by `numEntries <= 4`.

That's not quite right. Both leaf and non-leaf nodes will have `numEntries <= 4`. `numEntries` just indicate how many of the elements in the `entries` array are actually populated (it could be between 1 and 4 inclusive).

"Leaf" or "non-leaf" status is really a status on each entry, rather than on each node, because it's possible (although not the case right now) that some entries in a node could point to deeper nodes in the tree, and other entries could just contain a terminal record ID.

> But if that's not the case, perhaps still putting `func (e entry) isLeaf() bool { return e.child == nil }` might make it more readable, or would it be more confusing? 😅

I think it might end up just making things more confusing ("leaf" is a direct synonym for "has no child"). Looking through the code now, I've realised that some comments conflate entries/children in a bit of a confusing way, so I've fixed that up in a98c3fc7f4599bea623ee42322e2f3a2935ca953. Let me know if that helps 😄